networkx/algorithms/shortest_paths/generic.py

```diff
@@ -163,21 +163,23 @@ def shortest_path_length(G, source=None, target=None, weight=None):
 
     Returns
     -------
-    length: int or dictionary
+    length: int or iterator
         If the source and target are both specified, return the length of
         the shortest path from the source to the target.
 
-        If only the source is specified, return a dictionary keyed by
-        targets whose values are the lengths of the shortest path from the
-        source to one of the targets.
+        If only the source is specified, return a tuple
+        (target, shortest path length) iterator. Where shortest path lengths
```

Comment:
Where shortest path lengths is a fragment. I think it should be a continuation of the previous sentence.

`... iterator, where shortest path lengths are ...`